### DISCUSSION TITLE
Error HTTP 500 after last update v1.122.3

### DISCUSSION THREAD

**Discussion by herculespt** [immich-app/immich#14693] [General]
Hi,

Can you please shed some light? I have updated from 122.1 to this last one, haven't changed anything at all, I've been around since the beginning of Immich, always respected "breaking changes", "docker-compose" "environment" and so on. But I really don't catch what's this one about.

After updating, I got this message, and Immich server don't start. 

"Error response from daemon: Ports are not available: exposing port TCP 0.0.0.0:2283 -> 0.0.0.0:0: listen tcp4 0.0.0.0:22
83: bind: An attempt was made to access a socket in a way forbidden by its access permissions."

Thanks a lot for any help.

**Comment by bo0tzz** [comment]
We'll need some more detail, like what sort of system you're running on, your compose and env files, etc.

**Reply by herculespt** [reply]
thanks bo0tzz. 

I am running a dedicated pc with windows 10, docker desktop, default last docker-compose.yml and .env, but as I have told, I haven't changed anything.

But I have figured out a few minutes ago. Trying several things, and I found that for some reason windows has reserved and blocked the ports 2283 i've been using since always. Last resort was to do an IP Reset.  Even in safe mode or NT/Authority couldn't delete the excluded port ranges for those ones. A very strange issue.... Windows has some stupid stuff sometimes.

But thanks for your reply.

**Comment by herculespt** [comment]
well... after a couple of hours, same thing happens. it seems Docker Desktop has ports 2280 until 2379 reserved somehow. Even changing in docker-compose to another free port, like 3001:2283, localhost:3001 shows nothing

**Comment by herculespt** [comment]
ok. so, I found that desktop docker for windows, is now block and reserve the ports that Immich has changed to, in few updates ago. so the host port 2283 cannot be used for that purpose. After removing the the port reservation, Docker doesn't work. Get's shutdown and have issues. I have tried with a clean install of both, docker and immich, and the default 2283 still doesn't go through. I changed in .env to 3001:2283, and with the clean immich install worked fine.

Now i'm going to try to understand what happened with my current immich server that was running before the desktop docker update. Still, i am going to try with the older desktop docker version, to see if that is really the issue.

**Comment by p3terse** [comment]
since Google finds this post when searching for this error, just thought I'd add something which might help other windows users - struggled with this for a couple of days, then found some general info from Docker that the Windows NAT driver can hold ports and make them unavailable. for me this was resolved by stopping the container, and restarting the winnat service:

- stop the immich docker container
- Open a  command prompt (CMD) as Administrator, and type:
```
net stop winnat
net start winnat
```
- Restart the immich container


What's weird is that the port reservation/unavailability survived multiple reboots and restarts of Docker Desktop, but was resolved by cycling the winnat service, which is unexpected. Anyway, might help a future reader of this post...
